### PR TITLE
ADEN-5783 Add A9 video header bidding to Ooyala

### PR DESCRIPTION
**Pull request by v3rth**
## Links

* https://wikia-inc.atlassian.net/browse/ADEN-5783
* http://project43.wikia.com/wiki/SyntheticTests/Premium/FeaturedVideo?InstantGlobals.wgAdDriverA9BidderCountries=[XX]&InstantGlobals.wgAdDriverA9VideoBidderCountries=[XX]

## Description

Add support for A9 (Amazon) video header bidding and extract Ooyala ads to separated file

@Wikia/x-wing CR please
- is `new Promise` safe?
- is extract ad stuff to file ok?

### CODE REVIEW

**Review by nandy-andy — APPROVED** [review]

File: app/modules/ads.js [hunk 1]
```diff
@@ -552,6 +556,10 @@ class Ads {
 		}
 	}
 
+	waitForReady() {
```
Comment (kvas-damian) [inline app/modules/ads.js:559]: can we make public API more consistent?
`waitForReady()` which returns promise vs `onReady()` which accepts callback
Please decide on one and migrate the other.

File: app/modules/video-players/ooyala-video-ads.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		} else if (this.isA9VideoEnabled()) {
+			return this.parseBidderParameters()
+				.catch(() => {})
+				.then((additionalParams) => this.setupAdManager(additionalParams));
+		} else {
+			return this.setupAdManager();
+		}
+	}
+
+	setupAdManager(additionalParams) {
```
Comment (kvas-damian) [inline app/modules/video-players/ooyala-video-ads.js]: `additionalParams = {}`

File: app/modules/ads.js (hunk 1 above)
Comment (v3rth) [reply app/modules/ads.js:559]: Isn't better to keep it optional or use same method pass callback and return Promise?

Comment (kvas-damian) [reply app/modules/ads.js:559]: personally, I don't like if you have too many paths to achieve the same. Then you need to debug every path separately. But if you believe it's better for you then let it be.

**Review by nandy-andy — APPROVED** [review]

**Review by fraszczakszymon — APPROVED** [review]

**Comment by fraszczakszymon** [comment]
Wait for release https://github.com/Wikia/app/pull/13626

**Pull request merged**